### PR TITLE
[W5.11][W12-3] Low Gin Wee

### PR DESCRIPTION
**Pull request by LowGinWee**
Added Message object to person class.
Updated JUnit tests and I/O tests.
Updated UserGuide

### CODE REVIEW

**Comment by LowGinWee** [comment]
Ready for review

**Comment by lws803** [comment]
Very nice update on the person class and new creation of message class as well. Maybe you can update the developer docs as well to encourage developers to edit and learn how the message class works 👍

File: src/seedu/addressbook/data/person/Message.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return fullMessage;
+    }
+
+    @Override
+    public int hashCode() {
+        return fullMessage.hashCode();
```
Comment (CT15) [inline src/seedu/addressbook/data/person/Message.java:30]: What if fullMessage is null?

File: src/seedu/addressbook/data/person/Message.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            this.fullMessage = message.trim();
+        }else{
+            this.fullMessage = null;
+        }
+    }
+
+    public boolean exist(){
+        if(this.fullMessage == null || this.fullMessage == ""){
+            return false;
+        }else{
```
Comment (CT15) [inline src/seedu/addressbook/data/person/Message.java:18]: "else" here is redundant.
You can either use **guard clause** or **ternary operator**.

File: src/seedu/addressbook/commands/AddCommand.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import seedu.addressbook.data.person.Name;
 import seedu.addressbook.data.person.Person;
 import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.Message;
```
Comment (CT15) [inline src/seedu/addressbook/commands/AddCommand.java]: Try to arrange the import in alphabetical order.

File: src/seedu/addressbook/data/person/Message.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    public Message(String message) {
+        if(message != null && message != "") {
+            this.fullMessage = message.trim();
+        }else{
+            this.fullMessage = null;
+        }
+    }
+
+    public boolean exist(){
```
Comment (CT15) [inline src/seedu/addressbook/data/person/Message.java:15]: Does "doesExist()" sound like a better boolean method name?

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 5]
```diff
@@ -15,13 +15,7 @@
...
 import seedu.addressbook.data.AddressBook;
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
```
Comment (CT15) [inline test/java/seedu/addressbook/commands/AddCommandTest.java]: Imported classes should be listed explicitly

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 6]
```diff
@@ -24,12 +24,7 @@
 import seedu.addressbook.commands.ViewAllCommand;
 import seedu.addressbook.commands.ViewCommand;
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.*;
```
Comment (CT15) [inline test/java/seedu/addressbook/parser/ParserTest.java]: Imported classes should be listed explicitly

File: test/java/seedu/addressbook/storage/StorageFileTest.java [hunk 7]
```diff
@@ -18,6 +18,7 @@
 import seedu.addressbook.data.person.Email;
 import seedu.addressbook.data.person.Name;
 import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Message;
```
Comment (CT15) [inline test/java/seedu/addressbook/storage/StorageFileTest.java]: Not in alphabetical order

File: src/seedu/addressbook/storage/AddressBookDecoder.java [hunk 8]
```diff
@@ -11,12 +11,7 @@
 
 import seedu.addressbook.data.AddressBook;
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
```
Comment (CT15) [inline src/seedu/addressbook/storage/AddressBookDecoder.java]: Imported classes should be listed explicitly

**Comment by CT15** [comment]
You can close the PR once you are done reviewing.

**Pull request closed without merging**